src/builder.rs

```diff
@@ -73,6 +74,198 @@ impl ConfigMapBuilder {
     }
 }
 
+/// A builder for [`SecurityContext`] objects (not to be confused with `PodSecurityContext`).
+#[derive(Clone, Default)]
+pub struct SecurityContextBuilder {
+    security_context: SecurityContext,
+}
+
+impl SecurityContextBuilder {
+    /// Convenience function for a wide use-case.
+    pub fn run_as_root() -> SecurityContext {
+        SecurityContext {
+            run_as_user: Some(0),
+            ..SecurityContext::default()
+        }
+    }
+
+    pub fn new() -> SecurityContextBuilder {
+        SecurityContextBuilder::default()
+    }
+
+    pub fn allow_privilege_escalation(&mut self, value: bool) -> &mut Self {
+        self.security_context.allow_privilege_escalation = Some(value);
+        self
+    }
+
+    pub fn capabilities(&mut self, value: Capabilities) -> &mut Self {
+        self.security_context.capabilities = Some(value);
+        self
+    }
+
+    pub fn privileged(&mut self, value: bool) -> &mut Self {
+        self.security_context.privileged = Some(value);
+        self
+    }
+
+    pub fn proc_mount(&mut self, value: String) -> &mut Self {
+        self.security_context.proc_mount = Some(value);
+        self
+    }
+
+    pub fn read_only_root_filesystem(&mut self, value: bool) -> &mut Self {
+        self.security_context.read_only_root_filesystem = Some(value);
+        self
+    }
+
+    pub fn run_as_group(&mut self, value: i64) -> &mut Self {
+        self.security_context.run_as_group = Some(value);
+        self
+    }
+
+    pub fn run_as_non_root(&mut self, value: bool) -> &mut Self {
+        self.security_context.run_as_non_root = Some(value);
+        self
+    }
+
+    pub fn run_as_user(&mut self, value: i64) -> &mut Self {
+        self.security_context.run_as_user = Some(value);
+        self
+    }
+
+    pub fn se_linux_level(&mut self, level: &str) -> &mut Self {
+        self.security_context.se_linux_options =
+            Some(self.security_context.se_linux_options.clone().map_or(
```

Comment:
You can use [`.get_or_insert_default(Default::default)`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.get_or_insert_with) to get a `&mut T` from a `&mut Option<T>` without having to write down the logic twice or do any cloning.